api/cmd/portainer/main.go

```diff
@@ -471,6 +471,8 @@ func buildServer(flags *portainer.CLIFlags) portainer.Server {
 
 	proxyManager := proxy.NewManager(dataStore, digitalSignatureService, reverseTunnelService, dockerClientFactory, kubernetesClientFactory, kubernetesTokenCacheManager)
 
+	reverseTunnelService.ProxyManager = proxyManager
```

Comment:
seems a bit convoluted since `proxyManager` depends on `reverseTunnelService` and `reverseTunnelService` also depends on `proxyManager`.
Can we not make one the parent and another a child - such that these services do not rely on each other?

Reply:
I knew it. but it needs a good redesign and a huge refactor.